receiver/sqlserverreceiver/scraper.go

```diff
@@ -300,3 +329,231 @@ func (s *sqlServerScraperHelper) recordDatabaseStatusMetrics(ctx context.Context
+		}
+		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, rowsReturned, rowsReturnVal); cached {
+			record.Attributes().PutInt(DBPrefix+rowsReturned, diff)
+		}
+
+		// handling `total_logical_reads`
+		logicalReadsVal, err := strconv.ParseInt(row[logicalReads], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, logicalReads, logicalReadsVal); cached {
+			record.Attributes().PutInt(DBPrefix+logicalReads, diff)
+		}
+
+		// handling `total_logical_writes`
+		logicalWritesVal, err := strconv.ParseInt(row[logicalWrites], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, logicalWrites, logicalWritesVal); cached {
+			record.Attributes().PutInt(DBPrefix+logicalWrites, diff)
+		}
+
+		// handling `physical_reads`
+		physicalReadsVal, err := strconv.ParseInt(row[physicalReads], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		}
+		if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, physicalReads, physicalReadsVal); cached {
+			record.Attributes().PutInt(DBPrefix+physicalReads, diff)
+		}
+
+		// handling `execution_count`
+		totalExecutionCount, err := strconv.ParseInt(row[executionCount], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		} else {
+			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, executionCount, totalExecutionCount); cached {
+				record.Attributes().PutInt(DBPrefix+executionCount, diff)
+			}
+		}
+
+		// handle `total_worker_time`, storing milliseconds
+		workerTime, err := strconv.ParseInt(row[totalWorkerTime], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		} else {
+			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, totalWorkerTime, workerTime/1000); cached {
+				record.Attributes().PutInt(DBPrefix+totalWorkerTime, diff)
+			}
+		}
+
+		// handle `total_grant_kb`
+		memoryGranted, err := strconv.ParseInt(row[totalGrant], 10, 64)
+		if err != nil {
+			err = fmt.Errorf("row %d: %w", i, err)
+			errs = append(errs, err)
+		} else {
+			if cached, diff := s.cacheAndDiff(queryHashVal, queryPlanHashVal, totalGrant, memoryGranted); cached {
+				record.Attributes().PutInt(DBPrefix+totalGrant, diff)
+			}
+		}
+
+		// handling `query_text`
+		obfuscatedSQL, err := obfuscateSQL(row[queryText])
+		if err != nil {
+			s.logger.Error("failed to obfuscate query text", zap.Error(err))
+			errs = append(errs, err)
+		}
+		record.Attributes().PutStr(DBPrefix+queryText, obfuscatedSQL)
+
+		// handling `query_plan`
+		obfuscatedQueryPlan, err := obfuscateXMLPlan(row[queryPlan])
+		if err != nil {
+			s.logger.Error("failed to obfuscate query plan", zap.Error(err))
```

Comment:
same as above

Reply:
fixed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/37958/commits/9d926cd3bae0eb2c361c72a9b3f4190f8ee7a1b7